packages/block-editor/src/store/selectors.js

```diff
@@ -463,6 +463,68 @@ export function getBlockHierarchyRootClientId( state, clientId ) {
 	return parent;
 }
 
+/**
+ * Given a block client ID, returns the hierarchy tree of client ID.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block from which tree will be created.
+ *
+ * @return {Array} Hierarchy tree of client ID.
+ */
+export function getTree( state, clientId ) {
+	let current = clientId;
+	const tree = [ current ];
+	do {
+		current = state.blocks.parents[ current ];
+		if ( current ) {
+			tree.push( current );
+		}
+	} while ( current );
+	return tree;
+}
+
+/**
+ * Given a block client ID, returns the next element of the hierarchy from which the block is nested which should be selected onFocus, return the block itself for root level blocks.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block from which to find first to select client ID.
+ *
+ * @return {string} First to select client ID
+ */
+export function getFirstToSelectBlock( state, clientId ) {
```

Comment:
To make sure I understand correctly what this does, this is about finding the closest common ancestor to `clientId` and `selectedId`, right? So when I tap on `clientId`, this implements the logic of where the selection should go.

Reply:
Exactly.